From ba91b5e42943d2712ba20c8d1040d93863b43f8e Mon Sep 17 00:00:00 2001
From: Allen Winter <winter@kde.org>
Date: Tue, 6 Oct 2026 06:40:18 -0400
Subject: [PATCH] src/libical/CMakeLists.txt - add keyword PUBLIC to
 target_link_libraries (#1439)

With CMake v4.4.4 (or possibly earlier), target_link_libraries must
be either all-keyword or all-plain. Therefore, for any link library
not marked as PRIVATE we explicitly add then as PUBLIC.
---
 CHANGELOG.md               |  3 ++-
 src/java/CMakeLists.txt    |  2 +-
 src/libical/CMakeLists.txt | 10 +++++-----
 3 files changed, 8 insertions(+), 7 deletions(-)

--- src/java/CMakeLists.txt.orig
+++ src/java/CMakeLists.txt
@@ -44,7 +44,7 @@ target_include_directories(
     ${JNI_INCLUDE_DIRS}
 )
 
-target_link_libraries(ical_jni ical_cxx)
+target_link_libraries(ical_jni PUBLIC ical_cxx)
 
 if(NOT ANDROID)
   set_target_properties(
--- src/libical/CMakeLists.txt.orig
+++ src/libical/CMakeLists.txt
@@ -292,19 +292,19 @@ add_dependencies(ical ical-header)
 target_include_directories(ical INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>")
 
 if(DEFINED CMAKE_THREAD_LIBS_INIT)
-  target_link_libraries(ical ${CMAKE_THREAD_LIBS_INIT})
+  target_link_libraries(ical PUBLIC ${CMAKE_THREAD_LIBS_INIT})
 endif()
 if(ICU_FOUND)
   target_link_libraries(ical PRIVATE ${ICU_LIBRARIES})
   if(VCPKG_MANIFEST_MODE)
     if(NOT APPLE AND NOT WIN32)
-      target_link_libraries(ical "m;stdc++")
+      target_link_libraries(ical PUBLIC "m;stdc++")
     endif()
   endif()
 endif()
 
 if(WINCE)
-  target_link_libraries(ical ${WCECOMPAT_LIBRARIES})
+  target_link_libraries(ical PUBLIC ${WCECOMPAT_LIBRARIES})
 endif()
 
 if(MSVC)
@@ -362,9 +362,9 @@ if(LIBICAL_CXX_BINDINGS)
     ${icalcxx_LIB_SRCS}
   )
   add_dependencies(ical_cxx ical-header)
-  target_link_libraries(ical_cxx ical)
+  target_link_libraries(ical_cxx PUBLIC ical)
   if(DEFINED CMAKE_THREAD_LIBS_INIT)
-    target_link_libraries(ical_cxx ${CMAKE_THREAD_LIBS_INIT})
+    target_link_libraries(ical_cxx PUBLIC ${CMAKE_THREAD_LIBS_INIT})
   endif()
 
   if(MSVC)
